### PR TITLE
Preliminary support of Trino dialect

### PR DESCRIPTION
**Pull request by efung**
### Brief summary of the change made
Add a new dialect for [Trino](https://trino.io/).

Makes progress on #4296

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Included test cases to demonstrate any code changes, which may be one or more of the following:
  - [x] `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - [x] Added dialect to the list in the docs

### CODE REVIEW

**Comment by WittierDinosaur** [comment]
Implementation looks like a good starting point - looks like you need to run `black`

**Comment by greg-finley** [comment]
Some more CI failures just now.

File: test/fixtures/dialects/trino/trino_bare_functions.sql
Comment (greg-finley) [inline test/fixtures/dialects/trino/trino_bare_functions.sql:1]: After https://github.com/sqlfluff/sqlfluff/pull/4955, let's just call these two `bare_functions.sql` and `bare_functions.yml` (without the `trino_`)

**Comment by WittierDinosaur** [comment]
@efung would you like me to get this over the line?

**Comment by efung** [comment]
> @efung would you like me to get this over the line?

@WittierDinosaur Yes please, I don't have time to plug away on it, sadly!

File: test/fixtures/dialects/trino/trino_bare_functions.sql
Comment (alanmcruickshank) [reply test/fixtures/dialects/trino/trino_bare_functions.sql:1]: I've resolve this one 👍

**Review by greg-finley — APPROVED** [review]
🎉

**Review by alanmcruickshank — APPROVED** [review]
Looks like a great start!

**Pull request merged**